ldap/servers/slapd/log.c

```diff
@@ -243,30 +248,36 @@ slapd_log_error_proc_internal(
-        PRErrorCode prerr = PR_GetError();                                                                                                                   \
-        syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n", prerr, slapd_pr_strerror(prerr), (buffer) + (headersize)); \
-    }
-#define LOG_WRITE_NOW(fd, buffer, size, headersize, err)                                                                                                         \
-    do {                                                                                                                                                         \
-        (err) = 0;                                                                                                                                               \
-        if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {                                                                            \
-            PRErrorCode prerr = PR_GetError();                                                                                                                   \
-            syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n", prerr, slapd_pr_strerror(prerr), (buffer) + (headersize)); \
-            (err) = prerr;                                                                                                                                       \
-        }                                                                                                                                                        \
-        /* Should be a flush in here ?? Yes because PR_SYNC doesn't work ! */                                                                                    \
-        PR_Sync(fd);                                                                                                                                             \
+#define LOG_WRITE(fd, buffer, size, headersize)                                                   \
+    if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {                 \
+        PRErrorCode prerr = PR_GetError();                                                        \
+        syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n", \
+               prerr, slapd_pr_strerror(prerr), (buffer) + (headersize));                         \
+        rc = -1;                                                                                  \
+    }
+#define LOG_WRITE_NOW(fd, buffer, size, headersize, err)                                              \
+    do {                                                                                              \
+        (err) = 0;                                                                                    \
```

Comment:
Instead of having LOG_WRITE_NOW_NO_ERR, we could test (err !=0) in LOG_WRITE_NOW and only use LOG_WRITE_NOW.

Reply:
> Instead of having LOG_WRITE_NOW_NO_ERR, we could test (err !=0) in LOG_WRITE_NOW and only use LOG_WRITE_NOW.

I'm a little weary of changing the behavior too much as I found that if we check all errors then things break.  Basically all the changes are just copy and paste of how it worked before.  I just organized it better.  Perhaps in a  new ticket we can look at some of the changes you are proposing?